actionpack/test/controller/required_params_test.rb

```diff
@@ -22,17 +22,17 @@ class ActionControllerRequiredParamsTest < ActionController::TestCase
     end
   end
 
-  if defined?(DidYouMean) && DidYouMean.respond_to?(:correct_error)
+  if defined?(DidYouMean::SpellChecker)
     test "exceptions have suggestions for fix" do
       error = assert_raise ActionController::ParameterMissing do
         post :create, params: { magazine: { name: "Mjallo!" } }
       end
-      assert_match "Did you mean?", error.message
+      assert_no_match "Did you mean?", error.message
 
       error = assert_raise ActionController::ParameterMissing do
-        post :create, params: { book: { title: "Mjallo!" } }
+        post :create, params: { boko: { title: "Mjallo!" } }
       end
-      assert_match "Did you mean?", error.message
+      assert_match 'Did you mean?  "boko"', error.message
```

Comment:
I'm not sure if this is suggesting what it should be suggesting. Perhaps something we should iterate on or even remove it until we find a good solution.

Reply:
Sorry, I should have added the suggestion to the assert.
This should suggest name as a correction for title.
It tests that nested parameters work as well, but maybe doesn't add much.